### DISCUSSION TITLE
Retrieve assignment value

### DISCUSSION THREAD

**Discussion by Kludex** [Instagram/LibCST#772] [Q&A]
Hi there :wave: 

How do I retrieve the assignment value?

I have this rule:
```python
    @m.leave(
        m.Call(
            func=m.Attribute(
                attr=m.OneOf(*[m.Name(method) for method in CLIENT_METHODS])
            ),
            args=[m.ZeroOrMore(), m.Arg(keyword=m.Name("data")), m.ZeroOrMore()],
        )
    )
    def replace_data_by_content(
        self, original_node: cst.Call, updated_node: cst.Call
    ) -> cst.Call:
        scope = self.get_metadata(ScopeProvider, original_node)
        for assignment in scope.assignments:
            print(assignment.name) # When we get "data", I want to retrieve its value.
        args = []
        for arg in updated_node.args:
            if m.matches(arg, m.Arg(keyword=m.Name("data"))):
                args.append(arg.with_changes(keyword=cst.Name("content")))
            else:
                args.append(arg)
        return updated_node.with_changes(args=args)
```

Consider this source:

```python
from starlette.testclient import TestClient

def test(client: TestClient):
    data = b"potato"
    response = client.post("/", data=data)
```

I need this to turn into:

```python
from starlette.testclient import TestClient

def test(client: TestClient):
    data = b"potato"
    response = client.post("/", content=data)  # change "data" by "content"
```

We need to convert `data` by `content` **only** if the `data` variable is `bytes` or `string` (`SimpleString`), on other cases it should remain `data`. For that reason, I need to get the `'b"potato"'` `SimpleString`. Any way to do it?

Thanks! :pray:

**Comment by zsol** [comment]
`assignment` should have a [`node`](https://py.wtf/libcst/libcst.metadata.scope_provider/Assignment#libcst.metadata.scope_provider.Assignment.node) attribute. Is that enough?

**Reply by Kludex** [reply]
No, because I have only Name("data") there. 😟

**Reply by Kludex** [reply]
Ah, I'm in the wrong node... What's the best practice to get the data assignment? Match the Assign node and store assignments? 🤔

**Reply by zsol** [reply]
Hmm yeah, in that case you can either collect all `Assign` nodes, or just use `ParentNodeProvider` to go up the tree from that `Name("data")` node

**Reply by Kludex** [reply]
Oh! I'm going to try that. Thanks! 🙏

**Reply by Kludex** [reply]
The `ParentNodeProvider` didn't help much. :thinking:
